sdk/mosaic.go

```diff
@@ -152,30 +185,37 @@ func (dto mosaicPropertiesDTO) toStruct() *MosaicProperties {
 	)
 }
 
-func (ref *mosaicInfoDTO) setMosaicInfo() (*MosaicInfo, error) {
-	publicAcc, err := NewAccountFromPublicKey(ref.Mosaic.Owner, NetworkType(1))
+type mosaicContext struct {
```

Comment:
There is no point in implementing such pattern here. Just pass regular networkType to setMosaicInfo (or toStruct) method.